# NumPy vs Python: Natural Log

In [2]:
import math
import numpy as np

In [3]:
rng = np.random.default_rng()

In [11]:
array = np.array(rng.integers(low=1, high=10, size=(1000000, 10)))

In [12]:
%%timeit

for i in range(1000000):
    for j in range(10):
        math.log(array[i, j])

1.16 s ± 26.2 ms per loop (mean ± std. dev. of 7 runs, 1 loop each)


In [15]:
%%timeit

np.log(array)

43.3 ms ± 979 µs per loop (mean ± std. dev. of 7 runs, 10 loops each)


## Summary

### Python
- took 1.16 seconds on average, with 26.2 ms of standard deviation, or an average of 1160 ms

### NumPy ufunc
- 43.4 ms on average, with 979 µs of standard deviation

The Numpy function ran 26.7 times faster than with python commands